docs/content/docs/data-structures/tree.md

```diff
@@ -485,61 +498,98 @@ For the meaning of "simultaneously", see [Types of distributed data structures](
+
+An `event` object is automatically passed to the `listener`. It has one member:
 
 -   `event.target`: The node on which the event was triggered.
--   `event.isLocal`: Specifies whether the change was made on the local client or a remote client.
--   `event.stopPropagation()`: If called in the listener, it stops the event from being triggered on the parent, in the tree, of the `event.target`.
 
 The `Tree.on()` method returns a function that unsubscribes the handler from the event. This method is typically called in clean up code when the node is being removed.
 
-## Type guard
+### Type guard
 
-When your code needs to process nodes only of a certain type and it has a reference to an object of an unknown type, you can use the TypeScript `instanceOf` keyword to test for the desired type as in the following examples.
+When your code needs to process nodes only of a certain type and it has a reference to an object of an unknown type, you can use the `Tree.is()` method to test for the desired type as in the following examples.
 
 ```typescript
-if (myNode instanceOf sf.number) {
-   // Code here that processes number nodes.
-}
+Tree.is(someNode: SharedTreeNode, nodeType: TreeNodeSchema | T): boolean
+```
+
+Returns `true` if `someNode` is of type `nodeType`. Note that `T` is a type that is derived from a call of one of the `SchemaFactory` methods; `object()`, `map()`, or `array()`. Here are examples:
 
-if (myNode instanceOf Note) {
+```typescript
+if (Tree.is(myNode, Note)) {
    // Code here that processes Note nodes.
 }
 ```
 
-Alternatively, the `Tree` class provides a method for testing the type of a node:
+For another example, see the `Tree.parent()` method in [Node information](#node-information).
+
+### Transactions
+
+If you want the `SharedTree` to treat a set of changes atomicly, wrap these changes in a transaction. Using a transaction guarantees that all of the changes will be applied together or none of them will. Transactions will include the concept of a constraint in the future which will allow the changes in a transaction to fail if the constraint is not met.
+
+Further, changes within a transaction are exposed through a single `Revertible` object so that they can be reverted as a single change by the undo/redo code. See [Undo/Redo support](#undoredo-support) for more information about `Revertible` objects.
+
+To create a transaction use the `Tree.runTransaction()` method. You can cancel a transaction from within the callback function by returning the string "rollback". Also, if an error occurs within the callback, the transaction will be canceled.
+
+In this example, myNode can be any node in the SharedTree. It will be optionally passed into the callback function.
```

Comment:
Why optionally? Do we not always pass the node?

Reply:
You don't have to specify an argument for it. So it is optional in that sense.